pppd/plugins/radius/Makefile.linux

```diff
@@ -43,13 +43,13 @@ install: all
 	$(INSTALL) -c -m 444 pppd-radattr.8 $(MANDIR)
 
 radius.so: radius.o libradiusclient.a
-	$(CC) -o radius.so -shared radius.o libradiusclient.a
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
```

Comment:
Again, also add $(LIBS) in case there are platform or compatibility libs that are needed (`-lposix`, `-lpthread`, or whatever).